### ISSUE TITLE
Strange indentation error with if/else alignment

### ISSUE THREAD

**Issue by zmoazeni** [bbatsov/rubocop#1691]
Ruby code:

```ruby
bar = true

some_var = if bar
  "it's true!"
else
  "it's not true"
end

puts some_var
```

rubocop config:

```yaml
Style/ElseAlignment:
  Enabled: false

Lint/EndAlignment:
  Enabled: false
```

Error message:

```
foo.rb:4:3: C: Use 2 (not -9) spaces for indentation.
  "it's true!"
  ^^^^^^^^^
```

---

If I don't want to enforce if/else formats, how can I disable this without getting a warning?

**Comment by bquorning** [comment]
Yeah, that’s a bit weird. The warning comes from the `Style/IndentationWidth` cop, but it actually looks at the `Lint/EndAlignment` configuration in this case.

To fix it, you would either disable `Style/IndentationWidth`, or enable `Lint/EndAlignment` with the configuration
```yaml
Lint/EndAlignment:
  AlignWith: variable
```

**Comment by zmoazeni** [comment]
> To fix it, you would either disable Style/IndentationWidth

Wouldn't this disable indentation checking across the board? Ignoring stuff like:

```ruby
def foo
    1 # 4 spaces
end
```

> or enable Lint/EndAlignment with the configuration

Like I said, we don't want to enforce if/else alignment rules. If we choose `variable` then it doesn't allow:

```ruby
some_var =
  if bar
    "it's true!"
  else
    "it's not true"
  end
```

**Comment by jonas054** [comment]
I think the `variable` style of `Lint/EndAlignment` should be changed so that it accepts (enforces) alignment with the `if` keyword in the above example where there is a line break after `=`.

**Comment by bbatsov** [comment]
> I think the variable style of Lint/EndAlignment should be changed so that it accepts (enforces) alignment with the if keyword in the above example where there is a line break after =.

Definitely.

**Comment by patbl** [comment]
So is it not possible to allow both

```
some_var = if bar
             "it's true!"
           else
             "it's not true"
           end
```

and

```
some_var = if bar
  "it's true!"
else
  "it's not true"
end
```

with `Layout/IndentationWidth` enabled?

**Comment by jonas054** [comment]
No. Allowing two styles is generally not something we like to do in RuboCop. It's either/or.
